src/anyp/TrafficMode.h

```diff
@@ -73,12 +73,93 @@ class TrafficMode
      */
     bool tunnelSslBumping = false;
 
-    /** true if the traffic is in any way intercepted
-     *
-     */
-    bool isIntercepted() { return natIntercept||tproxyIntercept ;}
+    PortKind portKind; ///< the parsed port type value
 };
 
+/**
+ * Set of 'mode' flags defining types of traffic which can be received.
+ *
+ * Use to determine the processing steps which need to be applied
+ * to this traffic under any special circumstances which may apply.
+ */
+class TrafficMode
+{
+public:
+    explicit TrafficMode(const TrafficModeFlags::PortKind aPortKind) : flags_(aPortKind) {}
+
+    /// This is a gateway (a.k.a. reverse proxy) port. TODO: Rename.
+    bool accelSurrogate() const { return flags_.accelSurrogate; }
+
+    /// This port handles intercepted traffic. Traffic may be intercepted many
+    /// times (at various locations) before reaching this port, including:
+    /// * between the user agent and the connecting TCP client and/or
+    /// * between the connecting TCP client and this Squid port.
+    /// This port mode alone does not imply that the client of the accepted TCP
+    /// connection was not connecting directly to this port (since commit
+    /// 151ba0d). In other words, we may not be an interception proxy ourselves.
+    bool interceptedSomewhere() const { return flags_.natIntercept || flags_.tproxyIntercept || proxySurrogateHttpsSslBump(); }
+
+    /// The user agent was explicitly configured to use a proxy at this port. We
+    /// are configured to assume that no interception has happened on the way.
+    /// This port configuration is also known as a forward proxy.
+    bool explicitProxy() const { return !interceptedSomewhere() && !accelSurrogate(); }
```

Comment:
@yadij, this PR needs polishing, but I suggest that we resume by trying to get an agreement regarding the above three primary TrafficMode methods while ignoring other problems. If we can agree on these three, it would be much easier to make further progress.

1. The first method -- accelSurrogate() -- inherited an ugly name from its flag. We are changing every existing user of this formally raw flag, so we can rename without making the diff a lot more complex. [HTTPbis](https://www.ietf.org/archive/id/draft-ietf-httpbis-semantics-19.html) says this is a "gateway" port, but they also acknowledge the "reverse proxy" name. If you want to leave this method name as is, fine. If you prefer to use one of the HTTPbis terms instead, let's use those! Perhaps there are even better options.
\
My concern here is whether an accelSurrogate port can accept PROXY protocol traffic where a user agent was talking to a haproxy instance that was pretending to be an origin server. If that is possible (and your commit 151ba0d seems to suggest that) then we should adjust documentation to say that explicitly, similar to what we did for interceptedSomewhere() -- we may not be a true/classic reverse proxy, but we assume that the user agent is talking to (what it believes to be) an origin server and no interception was happening somewhere between that user agent and us.

2. The middle method -- interceptedSomewhere() -- is the stumbling block of several change requests. I think the current name is adequate but please rename the way you want this method to be named. Please note the important caveat that since your commit 151ba0d, this kind of port is not necessarily a port where the last interceptor redirects its traffic. The PR code preserves that important semantics.

3. The last method -- explicitProxy() -- does not seem to be controversial. It now uses the name you suggested. Its meaning is fairly simple as well -- it corresponds to the "default" port configuration where the primary traffic handling mode (or whatever you want to call it) has _not_ been configured.

BTW, Eduard or I can do all the renaming legwork -- just let us know your name nominations.